libcontainer/cgroups/fscommon/utils.go

```diff
@@ -54,22 +54,22 @@ func ParseUint(s string, base, bitSize int) (uint64, error) {
 	return value, nil
 }
 
-// ParseKeyValue parses a space-separated "name value" kind of cgroup
+// ParseKeyValue parses a space-separated "key value" kind of cgroup
 // parameter and returns its key as a string, and its value as uint64
-// (ParseUint is used to convert the value). For example,
+// (using [ParseUint] to convert the value). For example,
 // "io_service_bytes 1234" will be returned as "io_service_bytes", 1234.
 func ParseKeyValue(t string) (string, uint64, error) {
-	parts := strings.SplitN(t, " ", 3)
-	if len(parts) != 2 {
+	key, val, ok := strings.Cut(t, " ")
+	if !ok {
```

Comment:
This is not doing the same. The old code checks that only two spaces are there, while the new code can have a lot of spaces and it will just pull out "key". I think this is fine because ParseUint() will fail, but the error will not be so clear.

Shall we modify our ParseUint function to return more clear errors? This, of course, applies to lot of commits doing the same thing.

I understand this all comes from the kernel, so if you don't think this is worth doing it it's fine. I think it can't hurt :)

Reply:
You got it wrong a bit. The old code checks that there is no more than a single space, and errors out otherwise.

The new code doesn't care (i.e. `value` can have spaces), but as you said, it will fail below in `ParseUint`, so instead of

> line "something 1234  xxx" is not in key value format

the error will be something like

> strconv.ParseUint: parsing "1234 xxx": invalid syntax

So, it is doing the same thing -- the only difference is which error is returned.

I think that the error here (either the old or the new one) is extremely unlikely, as we're parsing kernel data, and this is why I don't think it's an issue. It would definitely be an issue if we are to parse data provided by a user.

Reply:
Oh, okay, I had the impression ParseUnit didn't print the string it failed to parse. Perfect then! :)